internal/tracinginterceptor/interceptor.go

```diff
@@ -177,12 +177,70 @@ func (i *Interceptor) CallOneway(ctx context.Context, req *transport.Request, ou
 
 // HandleStream implements interceptor.StreamInbound
 func (i *Interceptor) HandleStream(s *transport.ServerStream, h transport.StreamHandler) error {
-	panic("implement me")
+	req := s.Request()
+	parentSpanCtx, _ := i.tracer.Extract(i.propagationFormat, getPropagationCarrier(req.Meta.Headers.Items(), i.transport))
+	transportRequest := &transport.Request{
+		Caller:    req.Meta.Caller,
+		Service:   req.Meta.Service,
+		Procedure: req.Meta.Procedure,
+		Headers:   req.Meta.Headers,
+		Transport: req.Meta.Transport,
+	}
+
+	extractOpenTracingSpan := &transport.ExtractOpenTracingSpan{
+		ParentSpanContext: parentSpanCtx,
+		Tracer:            i.tracer,
+		TransportName:     s.Request().Meta.Transport,
+		StartTime:         time.Now(),
+		ExtraTags:         commonTracingTags,
+	}
+	_, span := extractOpenTracingSpan.Do(s.Context(), transportRequest)
+	defer span.Finish()
+	err := h.HandleStream(s)
+	return updateSpanWithErrorDetails(span, err != nil, nil, err)
 }
 
 // CallStream implements interceptor.StreamOutbound
 func (i *Interceptor) CallStream(ctx context.Context, req *transport.StreamRequest, out transport.StreamOutbound) (*transport.ClientStream, error) {
-	panic("implement me")
+	createOpenTracingSpan := &transport.CreateOpenTracingSpan{
+		Tracer:        i.tracer,
+		TransportName: i.transport,
+		StartTime:     time.Now(),
+		ExtraTags:     commonTracingTags,
+	}
+	_, span := createOpenTracingSpan.Do(ctx, &transport.Request{
+		Caller:    req.Meta.Caller,
+		Service:   req.Meta.Service,
+		Procedure: req.Meta.Procedure,
+		Headers:   req.Meta.Headers,
+		Transport: req.Meta.Transport,
+	})
+
+	// Inject span context into headers for tracing propagation
+	tracingHeaders := make(map[string]string)
+	if err := i.tracer.Inject(span.Context(), i.propagationFormat, getPropagationCarrier(tracingHeaders, i.transport)); err != nil {
+		span.LogFields(logFieldEventError, log.String("message", err.Error()))
+	} else {
+		for k, v := range tracingHeaders {
+			req.Meta.Headers = req.Meta.Headers.With(k, v)
+		}
+	}
+
+	clientStream, err := out.CallStream(ctx, req)
+	if err != nil {
+		if updateErr := updateSpanWithErrorDetails(span, false, nil, err); updateErr != nil {
+			i.log.Error("Failed to update span with error details", zap.Error(updateErr))
```

Comment:
I feel this log is not accurate  - function `updateSpanWithErrorDetails`returns the error as-is and never returns errors because of failing to update span with error details. 

We should just return `updateSpanWithErrorDetails(span, false, nil, err)`.